File: selenium/src/test/java/com/ytfc/blog/pages/user/ForgotNamePage.java
package com.ytfc.blog.pages.user;

import com.ytfc.blog.utility.Helper;
import com.ytfc.blog.forms.ForgotNameForm;
import com.ytfc.blog.pages.PageObject;

import org.openqa.selenium.WebDriver;

public class ForgotNamePage extends PageObject {
    private ForgotNameForm form;

    public ForgotNameForm getForm() {
        return this.form;
    }

    public ForgotNamePage(WebDriver driver) {
        this.driver = driver;
        this.form   = new ForgotNameForm(driver);
    }

    public ForgotNamePage get() {
        Helper.get(driver, "user/forgot_name");

        assertPageTitle("Forgot name", "Page has correct title");
        form.assertFieldTypes("Field has correct type");

        return this; // Chainable
    }

    public void submit(String reqEmail) {
        get();
        form.getField("email").setValue(reqEmail);
        form.getButton().click();
    }

}

File: cypress/cypress/pages/forgot-name-page.js
import { Common         } from '../pages/common'
import { ForgotNameForm } from '../forms/forgot-name-form'
import { LoginPage      } from '../pages/login-page'

export class ForgotNamePage extends Common {
    constructor() {
        super()
        this._form = new ForgotNameForm()
    }

    visit() {
        new LoginPage().clickForgotNameLink()
        return this;
    }

    forgotName(email) {
        this.getForm().enter(new Map([['email', email]]))

        return this
    }

}
